tests/unit/test_option_parser.py

```diff
@@ -0,0 +1,187 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+""" Test the option parsing and filtering form the compilation commands. """
+
+import unittest
+
+from codechecker_lib import option_parser
+from codechecker_lib.option_parser import ActionType
+
+
+class OptionParserTest(unittest.TestCase):
```

Comment:
Right now the tests focus on the options which are critical to the analysis and needed to be parsed correctly. Further tests can be added later.
gcc/g++ and clang have many options we do not target all of them now. These options are continuously extended if needed.